es/es-server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -418,6 +418,79 @@ public void testSegments() throws Exception {
         }
     }
 
+    public void testSegmentsWithSoftDeletes() throws Exception {
```

Comment:
still not sure whether we have to backport soft-delete related changes, such as we do not use this feature.

Reply:
I wouldn't put effort into porting back non-used stuff currently. But as this is done now, lets merge it.